### PR TITLE
add setSpeed methode and change comments

### PR DESCRIPTION
**Pull request by YuvalRashman**

### CODE REVIEW

File: src/sensors/linearServo/LinearServo.java [hunk 1]
```diff
@@ -31,19 +31,42 @@ public LinearServo(int channel, int maxLength, int maxSpeed) {
...
+     *
+     *@param speed determine the direction
+     *
+     * speed equal to 0 stops the linear servo
+     * speed > 0 moves the linear servo forward
+     * speed < 0 moves the linear servo reverse
+     *
+     * dont forget to calibrate once the linearServo other wise it will not stop.
+     * */
+    public void setSpeed(double speed){
```
Comment (Romeore) [inline src/sensors/linearServo/LinearServo.java:60]: Conventions

File: src/sensors/linearServo/LinearServo.java [hunk 2]
```diff
@@ -31,19 +31,42 @@ public LinearServo(int channel, int maxLength, int maxSpeed) {
...
+    /** move the linear servo forward or reverse according to the speed sign
+     *
+     *@param speed determine the direction
+     *
+     * speed equal to 0 stops the linear servo
+     * speed > 0 moves the linear servo forward
+     * speed < 0 moves the linear servo reverse
+     *
+     * dont forget to calibrate once the linearServo other wise it will not stop.
+     * */
```
Comment (Romeore) [inline src/sensors/linearServo/LinearServo.java:59]: `Override` Annotation

File: src/sensors/linearServo/LinearServo.java [hunk 3]
```diff
@@ -31,19 +31,42 @@ public LinearServo(int channel, int maxLength, int maxSpeed) {
         currentPos = this.maxLength * this.get();
     }
 
+    /** calibrate the linear servo
+     * you can override this function and use different position or use setPosition once.
+     *
+     * the calibrate position doesn't meter and you only need to do this once*/
+    public void calibrate(){
```
Comment (Romeore) [inline src/sensors/linearServo/LinearServo.java:38]: Conventions

File: src/sensors/linearServo/LinearServo.java [hunk 4]
```diff
@@ -31,19 +31,42 @@ public LinearServo(int channel, int maxLength, int maxSpeed) {
...
+        setPosition(0);
+    }
+
     /**
      * @param setpoint the target position of the servo [mm]
      */
     public void setPosition(double setpoint) {
         setPos = MathUtil.clamp(setpoint, 0, maxLength);
-        setSpeed((setPos / maxLength * 2) - 1);
+        super.setSpeed((setPos / maxLength * 2) - 1);
```
Comment (Romeore) [inline src/sensors/linearServo/LinearServo.java:47]: I think we should put that calculation on function to clarify what we intend.

File: src/sensors/linearServo/LinearServo.java [hunk 5]
```diff
@@ -31,19 +31,42 @@ public LinearServo(int channel, int maxLength, int maxSpeed) {
...
+     *@param speed determine the direction
+     *
+     * speed equal to 0 stops the linear servo
+     * speed > 0 moves the linear servo forward
+     * speed < 0 moves the linear servo reverse
+     *
+     * dont forget to calibrate once the linearServo other wise it will not stop.
+     * */
+    public void setSpeed(double speed){
+        setPosition(speed == 0 ? currentPos : speed > 0 ? maxLength : 0);
```
Comment (Romeore) [inline src/sensors/linearServo/LinearServo.java:61]: I still dont agree about this way, but if you dont fix it next commit ill approve it..

**Review by Romeore — CHANGES_REQUESTED** [review]

File: src/sensors/linearServo/LinearServo.java [hunk 6]
```diff
@@ -32,18 +32,51 @@ public LinearServo(int channel, int maxLength, int maxSpeed) {
...
+     *
+     * speed equal to 0 stops the linear servo
+     * speed > 0 moves the linear servo forward
+     * speed < 0 moves the linear servo reverse
+     *
+     * dont forget to calibrate once the linearServo other wise it will not stop.
+     * */
+    @Override
+    public void setSpeed(double speed) {
+        setPosition(speed == 0 ? currentPos : Math.min(Math.max(Math.ceil(speed),0) * maxLength , maxLength));
```
Comment (Romeore) [inline src/sensors/linearServo/LinearServo.java:71]: You dont need Math.min,,

**Review by Romeore — CHANGES_REQUESTED** [review]

File: src/sensors/linearServo/LinearServo.java (hunk 6 above)
Comment (YuvalRashman) [reply src/sensors/linearServo/LinearServo.java:72]: It prevent from getting set point bigger from maxLength

File: src/sensors/linearServo/LinearServo.java [hunk 7]
```diff
@@ -32,18 +33,49 @@ public LinearServo(int channel, int maxLength, int maxSpeed) {
     }
 
     /**
+     * convert setpoint[mm] to PWM value
+     *
      * @param setpoint the target position of the servo [mm]
      */
-    public void setPosition(double setpoint) {
+    private double convertSetpointToPWM(double setpoint){
```
Comment (Romeore) [inline src/sensors/linearServo/LinearServo.java:40]: Conventions

File: src/sensors/linearServo/LinearServo.java [hunk 8]
```diff
@@ -78,5 +111,10 @@ public double getPosition() {
     public boolean isOnTarget() {
         return currentPos == setPos;
     }
+
+    /** disable motor and stop it */
+    public void disableLinearServo(){
```
Comment (Romeore) [inline src/sensors/linearServo/LinearServo.java:116]: Conventions

**Review by Romeore — CHANGES_REQUESTED** [review]

**Review by Romeore — APPROVED** [review]

**Pull request merged**